packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1643,6 +1685,20 @@ class InputDecorator extends StatefulWidget {
   /// How the text in the decoration should be aligned horizontally.
   final TextAlign textAlign;
 
+  /// {@template flutter.widgets.inputDecorator.textAlignVertical}
+  /// How the text should be aligned vertically.
+  ///
+  /// Determines the alignment of the baseline within the available space of
+  /// the input.  For example, TextAlignVertical.top will place the baseline
+  /// such that the text, and any attached decoration like prefix and suffix,
+  /// is as close to the top of the input as possible without overflowing.
+  /// Likewise, prefix and suffix height are included in the calculation at
```

Comment:
The heights of the prefix and suffix are similarly included for other alignment values.